include/deal.II/numerics/cell_data_transfer.h

```diff
@@ -0,0 +1,212 @@
+ *
+ * @code
+ * // prepare the triangulation,
+ * triangulation.prepare_coarsening_and_refinement ();
+ *
+ * // prepare the CellDataTransfer object for coarsening and refinement
+ * // and give the cell data vector that we intend to unpack later,
+ * Vector<double> data_to_transfer (triangulation.n_active_cells());
+ * //[fill data_to_transfer with cell-wise values...]
+ *
+ * CellDataTransfer<dim, spacedim, Vector<double>>
+ *   cell_data_trans (triangulation);
+ * cell_data_trans.prepare_for_coarsening_and_refinement ();
+ *
+ * // actually execute the refinement,
+ * triangulation.execute_coarsening_and_refinement ();
+ *
+ * // unpack transferred data,
+ * Vector<double> transferred_data (triangulation.n_active_cells());
+ * cell_data_trans.unpack (data_to_transfer, transferred_data);
+ * @endcode
+ *
+ * When using a parallel::shared::Triangulation, we need to ensure that we have
+ * the global data available in our local vector before refinement happened. We
+ * can achieve this as follows:
+ *
+ * @code
+ * Vector<double> data_to_transfer (triangulation.n_active_cells());
+ * //[fill data_to_transfer with cell-wise values...]
+ *
+ * PETScWrappers::MPI::Vector
+ * distributed_data_to_transfer (mpi_communicator,
+ *                               triangulation.n_active_cells(),
+ *                               triangulation.n_locally_owned_active_cells());
+ * for (const auto cell& : triangulation.active_cell_iterators ())
+ *   if (cell->is_locally_owned ())
+ *     {
+ *       const unsigned int index = cell->active_cell_index ();
+ *       distributed_data_to_transfer (index) = data_to_transfer (index);
+ *     }
+ * distributed_data_to_transfer.compress (VectorOperation::insert);
+ *
+ * data_to_transfer = distributed_data_to_transfer;
+ * @endcode
+ *
+ * For the parallel distributed case, a designated class
+ * parallel::distributed::CellDataTransfer is available. Please refer to this
+ * particular class when using a parallel::distributed::Triangulation.
+ *
+ * @note See the documentation of SolutionTransfer for matching code snippets
+ *   for transfer.
+ *
+ * @ingroup numerics
+ * @author Marc Fehling, 2019
+ */
+template <int dim, int spacedim = dim, typename VectorType = Vector<double>>
+class CellDataTransfer
+{
+private:
+  /**
+   * An alias that defines the data type of provided container template.
+   */
+  using value_type = typename VectorType::value_type;
+
+public:
+  /**
+   * Constructor.
+   *
+   * @param[in] triangulation The triangulation on which all operations will
+   *   happen. At the time when this constructor is called, the refinement
+   *   in question has not happened yet.
+   * @param[in] transfer_variable_size_data Specify whether your VectorType
+   *   container stores values that differ in size. A varying amount of data
+   *   may be packed per cell, if for example the underlying ValueType of
+   *   the VectorType container is a container itself.
```

Comment:
It doesn't look like `transfer_variable_size_data` is in this constructor. Should it be?

Reply:
Woops, I blindly copied that from the `p::d::CellDataTransfer` class. Thanks for noticing!